src/ticks/debugger_gdb.c

```diff
@@ -539,6 +544,47 @@ void debugger_step()
     debugger_active = 0;
 }
 
+uint8_t debugger_restore(const char* file_path, uint16_t at)
+{
+    FILE *f = fopen(file_path, "rb");
+    if (f == NULL)
+    {
+        printf("Could not open file.\n");
+        return 1;
+    }
+
+    size_t addr = at;
+
+    uint8_t buff[256];
+    size_t read_;
+    while ((read_ = fread(buff, 1, 256, f)))
+    {
+        char s[1024];
+        int offset;
+        sprintf(s, "M%zx,%zx:%n", addr, read_, &offset);
+        mem2hex(buff, &s[offset], read_);
+        const char* response = send_request(s);
+        if (strcmp(response, "OK") != 0)
+        {
+            printf("Warning: Cannot restore file at addr %zx: %s\n", addr, response);
+            return 1;
+        }
+
+        addr += read_;
+    }
+
+    fclose(f);
+
+    mem_requested_amount = 0;
+
+    // zero out all registers except for pc
+    struct debugger_regs_t regs;
```

Comment:
As I merged I thought about this bit.

Many of the CRTs expect a sensible stack value when entering - this sets the sp to top of memory which on the +zx isn't a great place - by default it comes in somewhere around the CLEAR value.

Reply:
Could you please fix that by doing `regs.sp = 0xFFFF;`?

Reply:
Oh now I see. Maybe we can find a symbol in debugging info to restore it from?

Reply:
I've just pushed a save/restore of sp over this operation. It's not perfect, but it does mean that sp stays roughly in the area it should be.

A user can always set sp manually should it have drifted to far as a result of multiple restores.

Reply:
> Oh now I see. Maybe we can find a symbol in debugging info to restore it from?

Normally sp on entry is inherited from BASIC etc - some targets set sp explicitly on entry, for those there's no problems.